software/NRG_itho_wifi/10_Gen_func.ino

```diff
@@ -551,3 +551,111 @@ void logInput(const char * inputString) {
 
 
 }
+
+void execAutoPilot()
+{
+  if (!systemConfig.autopilot_active)
+    return;
+
+  //
+  if (SHT3xupdated)
```

Comment:
First the MQTT tasks will be run (https://github.com/arjenhiemstra/ithowifi/blob/6504bdf108653eda8a4ab25560381a905b424196/software/NRG_itho_wifi/Loop.ino#L31)
if 'SHT3xupdated' is true it will be handled there and set to false again (https://github.com/arjenhiemstra/ithowifi/blob/6504bdf108653eda8a4ab25560381a905b424196/software/NRG_itho_wifi/Loop.ino#L55-L56) 
After that execAutoPilot() gets called and this check will always result false.

Reply:
Alright, got it. Didn't see that one coming. I used `SHT3xupdated` to reduce running code that won't do anything. Do you have another suggestion to replace it with or should we just discard this optimization?

Reply:
> Alright, got it. Didn't see that one coming. I used `SHT3xupdated` to reduce running code that won't do anything. Do you have another suggestion to replace it with or should we just discard this optimization?

For now I would just add a global boolean something like 'runAutoPilot' and set that to true after 'if (SHT3xupdated) {'. and use that as a trigger for the autopilot code. At least that will work.